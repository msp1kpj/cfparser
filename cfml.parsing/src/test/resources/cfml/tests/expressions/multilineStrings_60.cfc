component{

   function name(){
       var sql = "INSERT INTO thingTbl(
      thingId,
      name) VALUES (
      :thingId,
      :name";
   }
}