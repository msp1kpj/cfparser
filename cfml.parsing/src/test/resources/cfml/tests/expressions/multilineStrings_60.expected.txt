/*===TOKENS===*/
COMPONENT                <component>
'{'                      <{>
FUNCTION                 <function>
IDENTIFIER               <name>
'('                      <(>
')'                      <)>
'{'                      <{>
VAR                      <var>
IDENTIFIER               <sql>
'='                      <=>
OPEN_STRING              <">
STRING_LITERAL           <INSERT INTO thingTbl(
      thingId,
      name) VALUES (
      :thingId,
      :name>
CLOSE_STRING             <">
';'                      <;>
'}'                      <}>
'}'                      <}>
/*===TREE===*/
(scriptBlock
  (componentDeclaration
    component
    (componentGuts
      {
      (element
        (functionDeclaration
          function
          (identifier name)
          (
          parameterList
          )
          (compoundStatement
            {
            (statement
              (localAssignmentExpression
                var
                (multipartIdentifier (identifier sql))
                =
                (startExpression
                  (compareExpression
                    (baseExpression
                      (unaryExpression
                        (primaryExpression
                          (literalExpression
                            (stringLiteral
                              "
                              (stringLiteralPart
                                INSERT INTO thingTbl(\r\n      thingId,\r\n      name) VALUES (\r\n      :thingId,\r\n      :name
                              )
                              "
                            )
                          )
                        )
                      )
                    )
                  )
                )
              )
              ;
            )
            }
          )
        )
      )
      }
    )
  )
)
/*======*/
/*===DECOMPILE===*/
component {
  public function name()     {
      var sql = 'INSERT INTO thingTbl(
      thingId,
      name) VALUES (
      :thingId,
      :name';

    }
}
/*======*/